nav2_experimental/nav2_rl/nav2_turtlebot3_rl/navigator_dqn.py

```diff
@@ -0,0 +1,153 @@
+# Copyright (c) 2019 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from turtlebot3_env_core import TurtlebotEnv
+
+import rclpy
+from rclpy.node import Node
+from time import sleep
+
+import sys
+import numpy as np
+
+import parameters
+
+import gym
+from keras.models import Sequential
+from keras.layers import Dense, Activation, Flatten
+from keras.optimizers import Adam
+from rl.agents.dqn import DQNAgent
+from rl.policy import LinearAnnealedPolicy, BoltzmannQPolicy, EpsGreedyQPolicy
+from rl.memory import SequentialMemory
```

Comment:
I'm not familiar with this library, do you have a link?

Also are there rosdep keys for this? It should be added to the package.xml.

Reply:
Here is the link to the library: https://github.com/keras-rl/keras-rl.  You can use `pip install keras-rl` to instal. rosdep keys are there yet, I was actually planning to submit a PR to rosdistro to add it.

Reply:
Here is another library of interest that I'm planning to test. https://github.com/NervanaSystems/coach

Reply:
I never found any implementations I liked so I ended up making my own, but I never ran across these two.

Reply:
I have not verified these implementation, but since the second one is from Intel I'm planning to meet with them at some point and make it my default implementation.